samples/downloadPublicFile.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//   title: Storage Download Public File.
+//   description: Download Public File.
+//   usage: node downloadPublicFile.js <BUCKET_NAME> <SRC_FILE_NAME> <DEST_FILE_NAME>
+
+const path = require('path');
+const cwd = path.join(__dirname, '..');
+
+function main(
+  bucketName = 'my-bucket',
+  srcFileName = 'test.txt',
+  destFileName = path.join(cwd, 'downloaded.txt')
+) {
+  // [START storage_download_public_file]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const bucketName = 'Name of a bucket, e.g. my-bucket';
+  // const srcFileName = 'Remote file to download, e.g. file.txt';
+  // const destFileName = 'Local destination for file, e.g. ./local/path/to/file.txt';
+
+  const http = require('http');
+  const fs = require('fs');
+
+  function downloadPublicFile() {
+    const file = fs.createWriteStream(destFileName);
+
+    http.get(
+      `http://storage.googleapis.com/${bucketName}/${srcFileName}`,
+      response => {
+        response.pipe(file);
```

Comment:
is `response` a stream? [docs](https://nodejs.org/api/http.html#http_http_get_url_options_callback)
Isn't argument for `pipe()` needs to be a writable stream?

Reply:
Yes, `file` is writable stream.